demos/common/performance_metrics/metrics.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "metrics.h"
+
+#include <iostream>
+#include <limits>
+#include <sstream>
+#include <iomanip>
+
+PerformanceMetrics::PerformanceMetrics(Duration timeWindow) {
+    // 'timeWindow' defines the length of the timespan over which the 'current fps' value is calculated
+    timeWindowSize = timeWindow;
+
+    lastMovingStatistic = Statistic();
+    currentMovingStatistic = Statistic();
+    totalStatistic = Statistic();
+}
+
+void PerformanceMetrics::update(TimePoint lastRequestStartTime,
```

Comment:
I suggest you to split this method into two: `update` and `drawFps`. Or even make `drawFps` separated function.

Reply:
> I suggest you to split this method into two: `update` and `drawFps`. Or even make `drawFps` separated function.

We've discussed this earlier and the conclusion was:
> The idea behind merging methods is to prevent forgetting to call one of the methods as they are always expected to be called one after another in common use case. And to prevent calling that methods far form each other like
> ```
> performanceMetrics.update()
> postprocess_detections()
> performanceMetrics.put_moving_*()
> ```
From here https://github.com/openvinotoolkit/open_model_zoo/pull/1222#discussion_r442113416

So, I'm not sure that this method should be splitted.